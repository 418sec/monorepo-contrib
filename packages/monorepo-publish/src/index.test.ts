import { publish } from './index'

jest.mock('./util/exec', () => {
  const { execSync: _execSync } = require('child_process')
  return {
    calls: [] as { command: string }[],
    execSync(cwd: string, command: string) {
      this.calls.push({ command })
      if (!command.includes('lerna')) {
        return _execSync(command, { cwd })
      }
      // TODO: mock git clone
      return _execSync('echo', { cwd })
    },
  }
})

describe('publish', () => {
  it('should publish a project', () => {
    const execUtil = require('./util/exec')
    execUtil.calls = []

    const preRepkg = jest.fn(),
      prePublish = jest.fn(),
      postRepkg = jest.fn(),
      postPublish = jest.fn()

    // TODO: have publish return packages and snapshot that value
    publish(
      'https://github.com/facebookincubator/create-react-app.git',
      'HEAD',
      {
        exclude: ['create-react-app'],
        preRepkg,
        prePublish,
        postRepkg,
        postPublish,
      }
    )

    expect(execUtil.calls).toMatchSnapshot()
    expect(preRepkg.mock.calls.length).toBe(1)
    expect(prePublish.mock.calls.length).toBe(1)
    expect(postRepkg.mock.calls.length).toBe(1)
    expect(postPublish.mock.calls.length).toBe(1)
  })
})
