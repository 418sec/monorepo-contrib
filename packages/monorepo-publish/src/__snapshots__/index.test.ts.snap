// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publish should publish a project 1`] = `
Array [
  Object {
    "command": "git clone https://github.com/facebookincubator/create-react-app.git .",
  },
  Object {
    "command": "git config user.name \\"monorepo-publish\\"",
  },
  Object {
    "command": "git config user.email \\"monorepo-publish@hostname.local\\"",
  },
  Object {
    "command": "git reset --hard HEAD",
  },
  Object {
    "command": "git add -A",
  },
  Object {
    "command": "git commit -m \\"chore: alias packages\\"",
  },
  Object {
    "command": "./node_modules/.bin/lerna publish --independent --exact --force-publish=* --cd-version=prepatch --preid=HEAD --yes --skip-git --npm-tag=contrib",
  },
]
`;
